src/hooks/api/tokens/useSetSpamPreference.ts

```diff
@@ -0,0 +1,30 @@
+import { useCallback } from 'react';
+import { graphql } from 'relay-runtime';
+
+import { SetSpamPreferenceInput } from '~/generated/operations';
+import { usePromisifiedMutation } from '~/hooks/usePromisifiedMutation';
+
+export default function useSetSpamPreference() {
+  const [setSpamPreference] = usePromisifiedMutation(
+    graphql`
+      mutation useSetSpamPreferenceMutation($input: SetSpamPreferenceInput!) {
+        setSpamPreference(input: $input) {
+          ... on SetSpamPreferencePayload {
+            tokens {
+              isSpamByUser
+            }
+          }
+        }
+      }
+    `
+  );
+
+  return useCallback(
+    async (input: SetSpamPreferenceInput) => {
+      await setSpamPreference({ variables: { input } });
+
+      // TODO: handle error cases
+    },
```

Comment:
we'll want to make sure this is updated optimistically so that the collection is immediately moved to the hidden folder. @tbezman can you point to any good precedents in the codebase?

Reply:
haha oh yeah i had a few ideas of how to do it but all of them felt janky—but just now reading the relay docs more and seeing `optimisticResponse`